checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleConfiguration.java

```diff
@@ -54,18 +54,18 @@
     private static final Logger LOG = LoggerFactory.getLogger(CheckstyleConfiguration.class);
 
     private final CheckstyleProfileExporter confExporter;
-    private final RulesProfile profile;
+    private final ActiveRules activeRules;
     private final org.sonar.api.config.Configuration conf;
     private final FileSystem fileSystem;
 
     public CheckstyleConfiguration(
             org.sonar.api.config.Configuration conf,
             CheckstyleProfileExporter confExporter,
-            RulesProfile profile,
+            ActiveRules activeRules,
```

Comment:
@muhlba91 , please help me to understand a problem.
I still see this class in api of sonarqube, I placed link to issue description, so class is not deprecated, content of it heavily deprecated. Why do you think we change c-tor type ?

Reply:
In short, SQ has removed the access to this API only for ScannerSide (https://docs.sonarqube.org/plugins/servlet/mobile?contentId=1442007#content/view/1442007). This is why it's still working for the exporter on the server side. As mentioned in the comment already, we are kinda abusing the exporter for generating the checkstyle file on scanner side, because an exporter shall not be available/instantiated on scanner side and, with deprecation of the old `ScannerSide` side annotation and introducing the new one for 7.6, we will get a problem once they remove the old one because then no `JavaResourceLocator` will be available on ScannerSide on which our exporter relies on. In fact, we would need to refactor the exporter to be only on server side and create a new class for writing the configuration on the scanner side (e.g. through helper methods/classes).
This is why we have to replace `RulesProfile` here as we are working on ScannerSide where it's not available anymore.

Reply:
> Should we create an issue for that?

sure, please be welcome.